geopandas/base.py

```diff
@@ -40,18 +40,19 @@ def _geo_op(this, other, op):
 # TODO: think about merging with _geo_op
 def _series_op(this, other, op, **kwargs):
     """Geometric operation that returns a pandas Series"""
-    null_val = False if op != 'distance' else np.nan
+    null_val = False if op not in ['distance', 'project'] else np.nan
```

Comment:
Discovered this bug while testing. Don't know if there are any other operations that need this exception.

Reply:
Is there already a test that covers this?

Reply:
Yes, if you revert that particular change (but keeping all others), you get the following error:
```
ben@tigger:~/Programs/geopandas$ pytest geopandas/tests/test_geom_methods.py 
============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-3.1.2, py-1.4.34, pluggy-0.4.0
rootdir: /home/ben/Programs/geopandas, inifile:
collected 49 items 

geopandas/tests/test_geom_methods.py .................................F...............

=================================== FAILURES ===================================
_________________________ TestGeomMethods.test_project _________________________

self = <geopandas.tests.test_geom_methods.TestGeomMethods instance at 0x7f1cc8e76f80>

    def test_project(self):
        expected = Series([2.0, 1.5], index=self.g5.index)
        p = Point(1.0, 0.5)
>       self._test_binary_real('project', expected, self.g5, p)

geopandas/tests/test_geom_methods.py:415: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
geopandas/tests/test_geom_methods.py:114: in _test_binary_real
    *args, **kwargs)
geopandas/tests/test_geom_methods.py:173: in _binary_op_test
    fcmp(result, expected)
../../miniconda/lib/python2.7/site-packages/pandas/util/testing.py:1238: in assert_series_equal
    assert_attr_equal('dtype', left, right)
../../miniconda/lib/python2.7/site-packages/pandas/util/testing.py:1008: in assert_attr_equal
    raise_assert_detail(obj, msg, left_attr, right_attr)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

obj = 'Attributes', message = 'Attribute "dtype" are different'
left = dtype('bool'), right = dtype('float64'), diff = None

    def raise_assert_detail(obj, message, left, right, diff=None):
        if isinstance(left, np.ndarray):
            left = pprint_thing(left)
        if isinstance(right, np.ndarray):
            right = pprint_thing(right)
    
        msg = """{obj} are different
    
    {message}
    [left]:  {left}
    [right]: {right}""".format(obj=obj, message=message, left=left, right=right)
    
        if diff is not None:
            msg += "\n[diff]: {diff}".format(diff=diff)
    
>       raise AssertionError(msg)
E       AssertionError: Attributes are different
E       
E       Attribute "dtype" are different
E       [left]:  bool
E       [right]: float64

../../miniconda/lib/python2.7/site-packages/pandas/util/testing.py:1089: AssertionError
=============================== warnings summary ===============================
geopandas/tests/test_geom_methods.py::TestGeomMethods::()::test_intersection
  /home/ben/Programs/geopandas/geopandas/base.py:29: UserWarning: GeoSeries crs mismatch: None and {'init': 'epsg:4326', 'no_defs': True}
    other.crs))

-- Docs: http://doc.pytest.org/en/latest/warnings.html
=============== 1 failed, 48 passed, 1 warnings in 7.63 seconds ================
```

Prior to this change, `project` was returning the wrong thing for null values, but we didn't have a test for that (still don't actually). But at least my changes to the tests caught this dtype incorrectness. A more extensive overhaul of the testing system is needed, and I found it to be much more work than I had allotted for this problem.

Reply:
OK, thanks for the explanation!